crates/miden-testing/src/utils.rs

```diff
@@ -142,8 +143,16 @@ pub fn create_p2any_note(sender: AccountId, assets: &[Asset]) -> Note {
 ///
 ///  A `SPAWN` note contains a note script that creates all `output_notes` that get passed as a
 ///  parameter.
-pub fn create_spawn_note(sender_id: AccountId, output_notes: Vec<&Note>) -> anyhow::Result<Note> {
-    let note_code = note_script_that_creates_notes(output_notes);
+///
+/// # Errors
+///
+/// Returns an error if:
+/// - the sender of any of the provided output notes does not match `sender_id`.
+pub fn create_spawn_note<'note>(
+    sender_id: AccountId,
```

Comment:
I guess we could remove this and enforce consistency between each of the `output_notes`, right? In terms of API it might not be as clear though.